doc/source/io.rst

```diff
@@ -4806,6 +4806,55 @@ default ``Text`` type for string columns:
     Because of this, reading the database table back in does **not** generate
     a categorical.
 
+.. _io.sql.method:
+
+Insertion Method
+++++++++++++++++
+
+.. versionadded:: 0.24.0
+
+The parameter ``method`` controls the SQL insertion clause used.
+Possible values are:
+
+- ``'default'``: Uses standard SQL ``INSERT`` clause (one per row).
+- ``'multi'``: Pass multiple values in a single ``INSERT`` clause.
+  It uses a *special* SQL syntax not supported by all backends.
+  This usually provides better performance for analytic databases
+  like *Presto* and *Redshift*, but has worse performance for
+  traditional SQL backend if the table contains many columns.
+  For more information check the SQLAlchemy `documention
+  <http://docs.sqlalchemy.org/en/latest/core/dml.html#sqlalchemy.sql.expression.Insert.values.params.*args>`__.
+- callable with signature ``(pd_table, conn, keys, data_iter)``:
+  This can be used to implement a more performant insertion method based on
+  specific backend dialect features.
+
+Example of a callable using PostgreSQL `COPY clause
+<https://www.postgresql.org/docs/current/static/sql-copy.html>`__::
+
+  # Alternative to_sql() *method* for DBs that support COPY FROM
+  import csv
+  from io import StringIO
+
+  def psql_insert_copy(table, conn, keys, data_iter):
+      # gets a DBAPI connection that can provide a cursor
+      dbapi_conn = conn.connection
+      with dbapi_conn.cursor() as cur:
+          s_buf = StringIO()
+          writer = csv.writer(s_buf)
+          writer.writerows(data_iter)
+          s_buf.seek(0)
```

Comment:
Is there a reason you are not using `to_csv` here? It should be possible to write to a buffer with that as well.

Further, I think it would be good to actually run the code of this example, to make sure it works. We could either add it as a test case, or either actually run it in the documentation build (by applying it on a small example dataframe)

Reply:
My understanding is that `to_csv()` is a higher level function do a bunch of data conversions. 
At this point when this function is called `data_iter` has already passed trough pandas data conversions...

Reply:
Are we fixing this before merging? I can push a commit if we want.

Reply:
Started out as that, but this is kind of a can of worms to actually run this example. We would need to actually create a Postgres database. I'll see if something similar exists for sqlite.

Reply:
Actually nevermind, this is tested below and we don't actually have a DataFrame here we have a `chunk_iter`. It's not clear to me, but that seems to be a list of arrays, one per column, chunk pair?

Reply:
So +1 to merging as is.